packages/ui/src/accounts/components/SelectAccount/SelectAccount.tsx

```diff
@@ -33,6 +33,7 @@ export const SelectAccount = React.memo(({ onChange, filter, selected }: Props)
   const filteredOptions = useMemo(() => filterByText(options, search), [search, options])
   const keyring = useKeyring()
 
+  useEffect(() => setSelectedOption(selected), [selected])
```

Comment:
The changes broke the Account selection, but I don't understand how it use to work without this line. Am I missing something ?

Reply:
Previously the idea was to allow passing a pre-defined account to the component. It is used in line 30 as a default selected option in `useState()`.

Reply:
So there are two options
- passing account as `selected` -> the component is rendered with an already selected item
- passing `undefined` as `selected` -> the component is rendered as empty

Reply:
I get this, but I'm confused because it seems that after the component has been mounted: `setSelectedOption()` was only called when `filteredOptions.length === 0`. So I don't understand how the value of `selectedOption` updated, unless the component got unmounted

Reply:
The hook below (line 37) should be rewritten AFAICS, as it is hard to parse. _I think_ that it is used to set "unknown" account when user pastes an address to the search box which is not a known account.